model/src/main/scala/org/broadinstitute/dsde/rawls/StringValidationUtils.scala

```diff
@@ -30,6 +32,16 @@ trait StringValidationUtils {
     }
   }
 
+  private lazy val billingProjectNameRegex = "[A-z0-9_-]{6,30}".r
```

Comment:
could also move this into `UserService`, I left it here with the other string validators, but it's a little different as the only one to return a `Future` so if you would rather see it closer to the project creation code that uses it I'd be fine with that